Heart drop enemies

CameroonD opened the pull request:
Applied heart drop script to enemies
Rotated heart prefabs

CameroonD committed:
Rotated hearts 90 degrees

CameroonD committed:
Added Heart Drop script to enemies.

AleksandrHovhannisyan reviewed (dismissed):
Good work :)

One thing to keep in mind: full hearts currently only heal the player if the player has room for it.

For example, if you've taken only half a heart of damage so far, you should still be able to consume a full heart container to gain only half a heart of health.

Basically, all you'd need to check is if the player's lost health (which is `MaxHealth - Health`) is at least 1 (equivalent to half a heart). If it is, then you should apply either the entirety of the heart's value or the player's lost health, whichever is smaller.

In other words, you'd simply need to do the following in the heart script when applying its value to the player:

1. First get the player's lost health: `lostHealth = player.MaxHealth - player.Health`
2. Then, do `player.ChangeHealthBy(min(lostHealth, heart.value))`.

For example, if the player has lost 4 HP (2 hearts) and they touch a full heart (2 HP), then the minimum of those two is 1 HP. So the player would gain 2 HP. If, however, the player has lost only 1 HP (half a heart) and touches a full heart (equivalent of 2 HP), then the player will gain the 1 HP (half a heart).

CameroonD committed:
You can now consume whole hearts when only lacking half a heart
Increased hitbox of heart prefabs so they still trigger even after being flipped

AleksandrHovhannisyan reviewed (approved):
Looks good!
